pypesto/storage/hdf5.py

```diff
@@ -5,6 +5,31 @@
 import numpy as np
 
 
+def write_array(f: h5py.Group,
+                path: str,
+                values: Collection) -> None:
+    """
+    Write array to hdf5
+
+    Parameters
+    -------------
+    f:
+        h5py.File
+    path:
+        path of the dataset to create
+    values:
+        array to write
+    """
+
+    if all(isinstance(x, int) for x in values):
```

Comment:
I think here may be potential problems. E.g. `np.float64` is no subclass of `float`. Generic types may be `numbers.{Float,Integral}`, not sure.

Reply:
> E.g. `np.float64` is no subclass of `float`

Disagreeing there. I was also recently surprised about that... 
```
In [5]: issubclass(np.float64, float)                                                                                                                                                                                  
Out[5]: True
```

Reply:
wow. remember having tons of problems here at some point. did that change? anyway, so looks like xzy64 is subclassing xzy always. Instances where one uses e.g. float32 or int8 should be rare, so this might work ...

Reply:
But you are right, as this is only true for float64, not for others. Dunno if there are different python versions around with different default float type.

Agreed that `numbers.{Float,Integral}` should be used.